pkg/intermediate/aggregate.go

```diff
@@ -215,6 +215,71 @@ func (a *AggregationProcess) GetExpiryFromExpirePriorityQueue() time.Duration {
 	return a.inactiveExpiryTimeout
 }
 
+// GetRecordStrings returns the flow records by FlowKey from flowKeyRecordMap.
+// Returns all the flow records if FlowKey is not provided.
+func (a *AggregationProcess) GetRecordStrings(flowKey *FlowKey) []string {
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+
+	var records []string
+	for currentFlowKey, record := range a.flowKeyRecordMap {
+		if flowKey != nil {
+			if (flowKey.SourceAddress != "" && flowKey.SourceAddress != currentFlowKey.SourceAddress) ||
+				(flowKey.DestinationAddress != "" && flowKey.DestinationAddress != currentFlowKey.DestinationAddress) ||
+				(flowKey.Protocol != 0 && flowKey.Protocol != currentFlowKey.Protocol) ||
+				(flowKey.SourcePort != 0 && flowKey.SourcePort != currentFlowKey.SourcePort) ||
+				(flowKey.DestinationPort != 0 && flowKey.DestinationPort != currentFlowKey.DestinationPort) {
```

Comment:
Could we do a hash check before checking individual elements to improve the performance?

Reply:
Updated